typescript/core/src/v0/Mcap0IndexedReader.test.ts

```diff
@@ -548,4 +548,280 @@ describe("Mcap0IndexedReader", () => {
+    chunk1.addMessage(message6);
+
+    const chunk2 = new ChunkBuilder({ useMessageIndex: true });
+    chunk2.addChannel(channel1);
+    chunk2.addMessage(message4);
+
+    const chunk3 = new ChunkBuilder({ useMessageIndex: true });
+    chunk3.addChannel(channel1);
+    chunk3.addMessage(message5);
+
+    const builder = new Mcap0RecordBuilder();
+    builder.writeMagic();
+    builder.writeHeader({ profile: "", library: "" });
+
+    const chunkIndexes: TypedMcapRecords["ChunkIndex"][] = [];
+    function writeChunk(chunk: ChunkBuilder) {
+      const chunkStartOffset = BigInt(builder.length);
+      const chunkLength = builder.writeChunk({
+        messageStartTime: chunk.messageStartTime,
+        messageEndTime: chunk.messageEndTime,
+        uncompressedSize: BigInt(chunk.buffer.length),
+        uncompressedCrc: 0,
+        compression: "",
+        records: chunk.buffer,
+      });
+
+      const messageIndexStart = BigInt(builder.length);
+      let messageIndexLength = 0n;
+      const chunkMessageIndexOffsets = new Map<number, bigint>();
+      for (const messageIndex of chunk.indices) {
+        chunkMessageIndexOffsets.set(
+          messageIndex.channelId,
+          messageIndexStart + messageIndexLength,
+        );
+        messageIndexLength += builder.writeMessageIndex(messageIndex);
+      }
+
+      chunkIndexes.push({
+        type: "ChunkIndex",
+        messageStartTime: chunk.messageStartTime,
+        messageEndTime: chunk.messageEndTime,
+        chunkStartOffset,
+        chunkLength,
+        messageIndexOffsets: chunkMessageIndexOffsets,
+        messageIndexLength,
+        compression: "",
+        compressedSize: BigInt(chunk.buffer.byteLength),
+        uncompressedSize: BigInt(chunk.buffer.byteLength),
+      });
+    }
+
+    writeChunk(chunk1);
+    writeChunk(chunk2);
+    writeChunk(chunk3);
+
+    builder.writeDataEnd({ dataSectionCrc: 0 });
+
+    const summaryStart = BigInt(builder.length);
+
+    for (const index of chunkIndexes) {
+      builder.writeChunkIndex(index);
+    }
+
+    builder.writeFooter({ summaryStart, summaryOffsetStart: 0n, summaryCrc: 0 });
+    builder.writeMagic();
+
+    {
+      const readable = makeReadable(builder.buffer);
+      const reader = await Mcap0IndexedReader.Initialize({ readable });
+      expect(readable.readCalls).toEqual(4);
+
+      const messageIter = reader.readMessages();
+      expect(readable.readCalls).toEqual(4);
+
+      const collected = [];
+
+      collected.push((await messageIter.next()).value);
+      expect(readable.readCalls).toEqual(6);
+      collected.push((await messageIter.next()).value);
+      expect(readable.readCalls).toEqual(8);
+      collected.push((await messageIter.next()).value);
+      expect(readable.readCalls).toEqual(10);
+      collected.push((await messageIter.next()).value);
+      expect(readable.readCalls).toEqual(10);
+
+      expect(collected).toEqual([message3, message4, message5, message6]);
+    }
+
+    {
```

Comment:
Could we dedupe these tests using `it.each([true, false])("", (reverse) => ...` or `for (const reverse of [true, false])` inside the test?